### ISSUE TITLE
[Feature] Horizontal Form Labels

### ISSUE THREAD

**Issue by Fagiolia1** [mobi/goponents#528]
### What is the feature that you would like to propose?
We would like to have a component created for prefix/suffix form labels. These labels would be both text and icons

### Describe the solution you'd like
Here is DSM with the horizontal form labels. This still needs a little more tweaking for the white design:
https://tangoe.invisionapp.com/dsm/tangoe/tangoe-ds/folder/components/5e6681d6d784abdb0e919667

_**Example:**_
Example of text input
![image](https://user-images.githubusercontent.com/8883257/78147453-885ff200-7401-11ea-8940-f7f54517e424.png)

**Comment by eligolding** [comment]
Can this be considered a dupe of #124?

**Comment by grahamhency** [comment]
@eligolding yes it is, but I think this issue provides a better visual for the component so I'll close #124.

Something I want to add as a note to this task though is that this should have the criteria of:
- Should work with any "inline" form input
- Should work with text or icon "label" on either end of the input

**Comment by grahamhency** [comment]
Probably should figure out #591 first before picking this up.

**Comment by grahamhency** [comment]
@mobi/tangoe-ux-ui I want to clarify on this issue which form components we would like this capability on?
- [ ] Datepicker
- [x] Input
- [x] Select
- [ ] Textarea
- [ ] Timepicker

**Comment by grahamhency** [comment]
Also we will need to consider a `dark` theme for these as well because all of our form elements do have a dark theme.

**Comment by kpmurphh** [comment]
> @mobi/tangoe-ux-ui I want to clarify on this issue which form components we would like this capability on?
> 
> * [ ]  Datepicker
> * [x]  Input
> * [x]  Select
> * [ ]  Textarea
> * [ ]  Timepicker

I would say the date and time pickers would benefit from this option as well if possible. I don't think textarea needs it. 

I'll add examples for both light and dark theme horizontal form labels to our DSM.

**Comment by grahamhency** [comment]
@kpmurphh let me know when you've got those documented and we'll be ready to start this task.
